internal/auth/authenticator_oidc.go

```diff
@@ -288,16 +345,28 @@ func (oa *OIDCAuthenticator) handleOAuth2Callback(tmpl *template.Template, error
 		}
 
 		var state State
-		err = json.Unmarshal(stateJSONPayload, &state)
+		err = msgpack.Unmarshal(stateJSONPayload, &state)
 		if err != nil {
 			logrus.Errorf("unable to unmarshal the state payload: %v", err)
 			http.Error(w, "Bad request", http.StatusBadRequest)
 			return
 		}
 
-		signatureOk := oa.signatureComputer.VerifySignature(state.URL, state.Signature)
-		if !signatureOk {
-			err = errorTmpl.Execute(w, struct{ URL string }{URL: string(state.URL)})
+		if state.Timestamp.Add(30 * time.Second).Before(time.Now()) {
```

Comment:
What is this delay? Is it worth adding a constant?

Reply:
This delay is the expiration of the state. Now, that you mention it, I will probably use JWT to represent the state since it would embed all required features.

Reply:
However, your question will remain regarding the constant. Do you want me to make it configurable? Is it really worth it at this stage?
This expiration is set to avoid replay attacks with the state. Basically, a hacker would have a 30 seconds window to replay the auth request given she obtains the auth code.

Reply:
The idea was not to make it configurable right now but simply to move it within a constant.
Sounds to me that this is a better pattern to define such things elsewhere.

Reply:
I agree. I will do that then and implement JWT in another review to not mix everything up.

Reply:
done